lib/internal/Magento/Framework/Data/Collection.php

```diff
@@ -492,6 +492,7 @@ public function clear()
     {
         $this->_setIsLoaded(false);
         $this->_items = [];
+        $this->invalidateSize();
```

Comment:
There is no need in this method to exist, especially introduce a protected one.

Simply assign null in both cases

```suggestion
        $this->_totalRecords = null;
```

Reply:
@orlangur thank you for noticing!

The idea was to call this `invalidateSize` in `\Magento\Framework\Data\Collection\AbstractDb::addFieldToFilter`,
in the same way as it's done for EAV collection:
`\Magento\Eav\Model\Entity\Collection\AbstractCollection::addAttributeToFilter`.

Probably I missed this point while performing the review.
@sergeynezbritskiy, could you please check it?

Reply:
@orlangur @dmytro-ch Sure, I'll check

Reply:
Hey guys, this issue has been fixed